accord-core/src/main/java/accord/impl/SizeOfIntersectionSorter.java

```diff
@@ -0,0 +1,51 @@
+package accord.impl;
+
+import accord.api.TopologySorter;
+import accord.local.Node;
+import accord.topology.ShardSelection;
+import accord.topology.Topologies;
+import accord.topology.Topology;
+
+public class SizeOfIntersectionSorter implements TopologySorter
+{
+    public static final TopologySorter.Supplier SUPPLIER = new Supplier() {
+        @Override
+        public TopologySorter get(Topology topology)
+        {
+            return new SizeOfIntersectionSorter(new Topologies.Single(this, topology, false));
+        }
+
+        @Override
+        public TopologySorter get(Topologies topologies)
+        {
+            return new SizeOfIntersectionSorter(topologies);
+        }
+    };
+
+    final Topologies topologies;
+    SizeOfIntersectionSorter(Topologies topologies)
+    {
+        this.topologies = topologies;
+    }
+
+    @Override
+    public int compare(Node.Id node1, Node.Id node2, ShardSelection shards)
+    {
+        int maxShardsPerEpoch = topologies.maxShardsPerEpoch();
+        int count1 = 0, count2 = 0;
+        for (int i = 0, mi = topologies.size() ; i < mi ; ++i)
+        {
+            Topology topology = topologies.get(i);
+            count1 += count(node1, shards, i * maxShardsPerEpoch, topology);
+            count2 += count(node2, shards, i * maxShardsPerEpoch, topology);
+        }
+
+        // sort more intersections later
+        return count1 - count2;
+    }
+
+    private static int count(Node.Id node, ShardSelection shards, int offset, Topology topology)
+    {
+        return topology.foldlIntOn(node, (j, s, v) -> s.get(j) ? v + 1 : v, shards, offset, 0, 0);
```

Comment:
This is a very short function definition, and I find it clearer when it can fit in a small space. I would be open to renaming `s` to `shard` - would that suffice?

Reply:
Also it looks like `j` could be `i` here, presumably it's been moved from somewhere there was a clash.

Reply:
It's mostly just my lack of comfort with left fold in general. Renaming to shard helps thanks.